src/topic_training/src/Begin_Here.md

```diff
@@ -0,0 +1,122 @@
+
+Note that there are quite a few sections of code that are marked as "Optional." You do not need to include everything in the examples when creating your own code. They are simply showing one way of setting up a Publisher and Subscriber in the same file.
+
+One file will be written in C++, and the other will be in Python. I recommend discussing with one of the Software Leads to determine which part of the rover you would like to be assigned to. This will help you decide which language to focus on for this training. However, being familiar with both languages will not hurt.
+
+Go through one of the files, making sure you understand all the code and comments. Come back to this file once you feel ready to move on.
+
+## Running a Publisher and Subscriber
+Before you are able to run the example code, find the CMakeLists.txt file that is outside of the src directory but still inside of the topic_training directory. 
+
+This file is mostly commented out. Make sure these are uncommented:
+
+- find_package(catkin REQUIRED
+    roscpp
+    rospy
+    std_msgs
+    message_generation
+  )
+
+- catkin_package(
+    CATKIN_DEPENDS message_runtime
+  )
+
+- include_directories(
+    ${catkin_INCLUDE_DIRS}
+  )
+
+The C++ and the Python file both require these to work.
+
+If you chose C++, also uncomment these:
+
+- add_executable(Demo src/Demo.cpp)
+  add_executable(PlotDemo src/PlotDemo.cpp)
+  add_executable(MotorChallenge src/MotorChallenge.cpp)
+
+- add_dependencies(Demo 
+    ${${PROJECT_NAME}_EXPORTED_TARGETS} 
+    ${catkin_EXPORTED_TARGETS}
+  )
+  add_dependencies(PlotDemo 
+    ${${PROJECT_NAME}_EXPORTED_TARGETS} 
+    ${catkin_EXPORTED_TARGETS}
+  )
+  add_dependencies(MotorChallenge 
+    ${${PROJECT_NAME}_EXPORTED_TARGETS} 
+    ${catkin_EXPORTED_TARGETS}
+  )
+
+- target_link_libraries(Demo 
+    ${catkin_LIBRARIES}
+  )
+  target_link_libraries(PlotDemo
+    ${catkin_LIBRARIES}
+  )
+  target_link_libraries(MotorChallenge
+    ${catkin_LIBRARIES}
+  )
+
+If you chose Python, also uncomment this:
+
+- catkin_install_python(PROGRAMS src/Demo.py src/PlotDemo.py src/MotorChallenge.py
+    DESTINATION ${CATKIN_PACKAGE_BIN_DESTINATION}
+  )
+
+In future projects, you might need to use some of the other configurations, builds, and installs, but for the purposes of this training, you only need these for the example code to run.
+
+You can also check out the package.xml file, which you also may need to tailor for future projects. However, everything should already be set up correctly, so do not change anything.
+
+In the terminal, run the command *roscore*. In a new terminal window, cd into the catkin workspace. In this case, use *cd WRoverPlayground*. Then, use the commands *catkin_make -DCMAKE_EXPORT_COMPILE_COMMANDS=ON* and *source devel/setup.bash*. These commands adjust the catkin workspace based on the CMakeLists.txt and package.xml files. You must run these commands everytime you change these two files. 
```

Comment:
For code or commands, use literal markers like `this` (\`this\`).  This comment applies to all code examples or commands in this file.

Reply:
Fixed in 7ece46c.